app/src/main/java/se/healthrover/ui_activity_controller/voice_control/SpeechRecognition.java

```diff
@@ -138,6 +129,23 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
         super.onActivityResult(requestCode, resultCode, data);
     }
 
+    // Method to set up the connection to Dialogflow API
+    // using a private access key. Add your own API key under src/res/raw
+    private void connectDialogflow(){
+        try {
+            InputStream stream = getResources().openRawResource(R.raw.dialogflow_access_key);
+            GoogleCredentials credentials = GoogleCredentials.fromStream(stream);
+            String projectId = ((ServiceAccountCredentials)credentials).getProjectId();
+            SessionsSettings.Builder settingsBuilder = SessionsSettings.newBuilder();
+            SessionsSettings sessionsSettings = settingsBuilder.setCredentialsProvider(FixedCredentialsProvider.create(credentials)).build();
+            sessionsClient = SessionsClient.create(sessionsSettings);
+            session = SessionName.of(projectId, UUID);
+
+        } catch (Exception e) {
+            userInterfaceUtilities.showCustomToast(SpeechRecognition.this, "Can't access the Dialogflow API...");
```

Comment:
Can we refactor the toast message so it uses a string from the string.xml file instead of the hardcode one here?

Reply:
> Can we refactor the toast message so it uses a string from the string.xml file instead of the hardcode one here?

Sure, thing I am on it, nice catch!